src/test/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProviderTest.java

```diff
@@ -1,20 +1,76 @@
 package com.iota.iri.storage.rocksDB;
 
-import com.iota.iri.conf.Configuration;
-import com.iota.iri.model.Hash;
+import com.iota.iri.model.IntegerIndex;
 import com.iota.iri.model.Transaction;
-import com.iota.iri.storage.Tangle;
-import com.iota.iri.network.TransactionRequester;
-import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.storage.Indexable;
+import com.iota.iri.storage.Persistable;
+import com.iota.iri.utils.Pair;
+import org.apache.commons.io.FileUtils;
 import org.junit.*;
-import org.junit.rules.TemporaryFolder;
+import org.junit.runners.MethodSorters;
 
+import java.io.File;
 import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
-import static com.iota.iri.controllers.TransactionViewModelTest.getRandomTransactionTrits;
-
-/**
- * Created by paul on 3/4/17 for iri.
- */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 public class RocksDBPersistenceProviderTest {
+
+    private static RocksDBPersistenceProvider rocksDBPersistenceProvider;
+    private static String dbPath = "tmpdb", dbLogPath = "tmplogs";
+
+    @BeforeClass
+    public static void setUpDb() {
+        rocksDBPersistenceProvider = new RocksDBPersistenceProvider(dbPath, dbLogPath, 10000);
+        rocksDBPersistenceProvider.init();
+    }
+
+    @AfterClass
+    public static void destroyDb() {
+        rocksDBPersistenceProvider.shutdown();
+        FileUtils.deleteQuietly(new File(dbPath));
+        FileUtils.deleteQuietly(new File(dbLogPath));
+        rocksDBPersistenceProvider = null;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        rocksDBPersistenceProvider.clear(Transaction.class);
+    }
+
+    @Test
+    public void testDeleteBatch() throws Exception {
+        Persistable tx = new Transaction();
+        byte[] bytes = new byte[Transaction.SIZE];
+                Arrays.fill(bytes, (byte) 1);
+        tx.read(bytes);
+        tx.readMetadata(bytes);
+        List<Pair<Indexable, Persistable>> pairs = IntStream.range(1, 1000)
+                .mapToObj(i -> new Pair<>((Indexable) new IntegerIndex(i), tx))
```

Comment:
Until the `PersistenceProvider` interface is fixed we will have this ugly casting